docs/reference/sql/limitations.asciidoc

```diff
@@ -3,6 +3,46 @@
 [[sql-limitations]]
 == SQL Limitations
 
+[float]
+[[large-parsing-trees]]
+=== Large queries may throw `ParsingExpection`
+
+{es-sql} uses https://www.antlr.org/[`ANTLR`] to create https://en.wikipedia.org/wiki/Abstract_syntax_tree[abstract syntax trees]
```

Comment:
Since the audience is comprised of users, I would eliminate the technical stuff from the intro - in fact the whole sentence and rework it as "Extremely large queries can consume too much memory during parsing in which case the engine will abort parsing and throw an error - in these cases, consider reducing the query to a smaller size by potentially splitting it apart".

P.S. Technically ANTLR creates a parse tree (not an AST).